### PR TITLE
CI: Install all known dependencies for test-builds.sh on GitHub

### PR DESCRIPTION
**Pull request by rousskov**
Authored-by: Amos Jeffries <squid3@treenet.co.nz>

The apt-get build-dep trick makes us dependent on Ubuntu decisions, but
it saves us work, and we can always add more dependencies manually if we
discover holes in the Ubuntu-maintained list.

Co-authored-by: Alex Rousskov <rousskov@measurement-factory.com>

### CODE REVIEW

File: .github/workflows/default.yaml [hunk 1]
```diff
@@ -102,13 +102,10 @@ jobs:
       - name: Install prerequisite Linux packages
         if: runner.os == 'Linux'
         run: |
+          # required for "apt-get build-dep" to work
+          sudo sed --in-place -E 's/# (deb-src.*updates main)/  \1/g' /etc/apt/sources.list
           sudo apt-get --quiet=2 update
-          sudo apt-get --quiet=2 install libtool-bin
-          sudo apt-get --quiet=2 install libcppunit-dev
-          sudo apt-get --quiet=2 install libldap2-dev
```
Comment (rousskov) [inline .github/workflows/default.yaml:108]: In my [tests](https://github.com/measurement-factory/squid/actions/runs/3228261796/jobs/5284123736), `apt-get build-dep` does not install libldap2-dev explicitly (in GitHub Actions environment), but it does install enough for Squid to detect LDAP support, removing the need for this temporary workaround -- Squid build `--with-ldap` succeeds without this line.

```
checking for LIBLDAP... yes
checking for ldap.h... yes
checking for lber.h... yes
checking for mozldap/ldap.h... no
checking for LDAP... 1
checking for OpenLDAP... 1
checking for Sun LDAP SDK... 0
checking for Mozilla LDAP SDK... 0
checking for LDAP_OPT_DEBUG_LEVEL... 1
checking for LDAP_SCOPE_DEFAULT... 1
checking for LDAP_REBIND_PROC... 1
checking for LDAP_REBINDPROC_CALLBACK... 0
checking for LDAP_REBIND_FUNCTION... 0
checking for LDAP_URL_LUD_SCHEME... 0
```

**Comment by rousskov** [comment]
PR description:

```md
Authored-by: Amos Jeffries <squid3@treenet.co.nz>
```

As a side effect, this PR will also test `Authored-by` support we recently added to Anubis: When the PR gets to a staging commit, that commit should be authored by Amos rather than me, and the above PR description header should be removed from the staged commit message.

@yadij, using build-dep was your idea, so I am listing you as the primary author of this improvement and requesting your explicit review.

**Comment by kinkie** [comment]
FYI, https://github.com/kinkie/dockerfiles/blob/master/ubuntu-jammy/Dockerfile is the dockerfile used for the build farm on Ubuntu

**Comment by rousskov** [comment]
> FYI, https://github.com/kinkie/dockerfiles/blob/master/ubuntu-jammy/Dockerfile is the dockerfile used for the build farm on Ubuntu

FYI: Amos [wanted](https://github.com/squid-cache/squid/pull/1160/files#r988544696) to use `apt-get build-dep` rather than listing individual dependencies like that Dockerfile does (and like the initial/official workflow specs started to do). This PR implements that Amos idea.

Said that, we are not limited to `build-dep`, of course. If you know of any dependencies that the PR specs currently do not install, please let me know or add them yourself to the PR branch. As the PR title says, we want test-builds.sh to run with all known dependencies installed. I just do not know how to get to that state from `build-dep`.

FWIW, here are the packages that `apt-get update` updated and `apt-get build-dep` installed AFAICT based on the test [job](https://github.com/squid-cache/squid/actions/runs/3229016210/jobs/5285829068) [log](https://pipelines.actions.githubusercontent.com/serviceHosts/a0887df5-48dc-49cd-811e-0306a6543d63/_apis/pipelines/1/runs/26/signedlogcontent/6?urlExpires=2022-10-11T21%3A24%3A10.0643093Z&urlSigningMethod=HMACV1&urlSignature=4WQTtHrh4Xbe3V50QuAsUdklKR4SAKU0z%2FdTgKnMLh8%3D):

```
libnss-systemd:amd64 (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4)
systemd-sysv (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4)
libpam-systemd:amd64 (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4)
systemd (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4)
libsystemd0:amd64 (249.11-0ubuntu3.6) over (249.11-0ubuntu3.4)
autopoint (0.21-4ubuntu4)
libdebhelper-perl (13.6ubuntu1)
dh-autoreconf (20)
libarchive-zip-perl (1.68-1)
libsub-override-perl (0.09-2)
libfile-stripnondeterminism-perl (1.13.0-1)
dh-strip-nondeterminism (1.13.0-1)
dwz (0.14-1build2)
gettext (0.21-4ubuntu4)
intltool-debian (0.35.0+20060710.5)
po-debconf (1.0.21+nmu1)
debhelper (13.6ubuntu1)
libgssrpc4:amd64 (1.19.2-2)
libkdb5-10:amd64 (1.19.2-2)
libkadm5srv-mit12:amd64 (1.19.2-2)
libkadm5clnt-mit12:amd64 (1.19.2-2)
comerr-dev:amd64 (2.1-1.46.5-2ubuntu1.1)
krb5-multidev:amd64 (1.19.2-2)
libcap-dev:amd64 (1:2.44-1build3)
libcppunit-1.15-0:amd64 (1.15.1-4)
libcppunit-dev:amd64 (1.15.1-4)
libevent-2.1-7:amd64 (2.1.12-stable-1build3)
libgnutls-openssl27:amd64 (3.7.3-4ubuntu1.1)
libunbound8:amd64 (1.13.1-1ubuntu5.1)
libgnutls-dane0:amd64 (3.7.3-4ubuntu1.1)
libgnutlsxx28:amd64 (3.7.3-4ubuntu1.1)
libidn2-dev:amd64 (2.3.2-2build1)
libp11-kit-dev:amd64 (0.24.0-6build1)
libtasn1-6-dev:amd64 (4.18.0-4build1)
nettle-dev:amd64 (3.7.3-1build2)
libgnutls28-dev:amd64 (3.7.3-4ubuntu1.1)
libldap-dev:amd64 (2.5.13+dfsg-0ubuntu0.22.04.1)
libnfnetlink-dev (1.0.1-3build3)
libnetfilter-conntrack-dev:amd64 (1.0.9-1)
libpam0g-dev:amd64 (1.4.0-11ubuntu2)
libsasl2-dev (2.1.27+dfsg2-3ubuntu1)
libsystemd-dev:amd64 (249.11-0ubuntu3.6)
libtdb1:amd64 (1.4.5-2build1)
libtdb-dev:amd64 (1.4.5-2build1)
dh-apparmor (3.0.4-2ubuntu2.1)
libecap3:amd64 (1.0.1-3.2ubuntu4)
libecap3-dev:amd64 (1.0.1-3.2ubuntu4)
libkrb5-dev:amd64 (1.19.2-2)
libldap2-dev (2.5.13+dfsg-0ubuntu0.22.04.1)
```

File: .github/workflows/default.yaml (hunk 1 above)
Comment (yadij) [reply .github/workflows/default.yaml:108]: Oh. I see Ubuntu is transitioning from libldap2-dev to libldap-dev...
```
Setting up libldap-dev:amd64 (2.5.13+dfsg-0ubuntu0.22.04.1) ...
```

**Review by yadij — APPROVED** [review]

**Pull request closed without merging**